### PR TITLE
[PhpUnitBridge] Properly handle phpunit arguments for configuration file

### PR DESCRIPTION
**Pull request by biozshock**
| Q             | A
| ------------- | ---
| Branch?       | 4.3
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| License       | MIT

PhpUnitBridge should properly parse cli arguments for configuration file.
After fixing #34300 the PhpUnitBridge stopped recognizing short `-c` option.
Also original PHPUnit allows to pass a directory as configuration parameter and read from either phpunit.xml or phpunit.xml.dist

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
/cc @ro0NL maybe?

@biozshock, please mind the CI failures.

**Comment by nicolas-grekas** [comment]
> @biozshock, please mind the CI failures.

actually not!! this file uses a specific code style, you can ignore fabbot :)

**Comment by ro0NL** [comment]
> After fixing #34300 the PhpUnitBridge stopped recognizing short -c option.

how did it regognize `-c` before then? AFAIK #34300 was exactly about supporting that. Thought mayhbe im missing it.. as i only considered the usecase from PhpStorm's perspectivice which invokes using `--configuration /project/phpunit.xml.dist` (the path i configured).

File: src/Symfony/Bridge/PhpUnit/bin/simple-phpunit.php [hunk 1]
```diff
@@ -24,15 +24,47 @@
 
     static $phpunitConfig = null;
     if (null === $phpunitConfig) {
-        $opt = min(array_search('-c', $opts = array_reverse($argv), true) ?: INF, array_search('--configuration', $opts, true) ?: INF);
         $phpunitConfigFilename = null;
-        if (INF !== $opt && isset($opts[$opt - 1])) {
-            $phpunitConfigFilename = $opts[$opt - 1];
```
Comment (ro0NL) [inline src/Symfony/Bridge/PhpUnit/bin/simple-phpunit.php]: checking for a directory here, and try file_exists phpunit.xml or .dist to resolve the config file path isnt sufficient?

**Comment by biozshock** [comment]
> how did it regognize `-c` before then? AFAIK #34300 / #34305 was exactly about supporting that. Though maybe im missing it.. as i only considered the usecase from PhpStorm's perspectivice which invokes using `--configuration /project/phpunit.xml.dist` (the path i configured).
It was recognizing `-c path/phpunit.xml` which not supported by PHPUnit, because notation is not GetOpt compatible.

**Comment by biozshock** [comment]
is it good to be merged, or i need to fix something here?
/cc @ro0NL @nicolas-grekas

**Review by ro0NL — APPROVED** [review]
works for me :) (phpstorm integration as described in #34300 is still OK :+1: )

**Comment by biozshock** [comment]
@nicolas-grekas is there anything else need to be done before merging this into next release?

**Comment by fabpot** [comment]
Thank you @biozshock.

**Pull request merged**